beets/dbcore/query.py

```diff
@@ -30,6 +30,11 @@
 
 if TYPE_CHECKING:
     from beets.dbcore import Model
+    from beets.dbcore.db import AnyModel
+
+    P = TypeVar("P", default=Any)
+else:
+    P = TypeVar("P")
```

Comment:
Same question regarding the `default` here as above: Do we really need it; and if so, shouldn't it be a more specific type?

Reply:
This saves us from having to specify `Any` when we need to use the generic `FieldQuery` in type definitions, like

```py
    _queries: dict[str, type[FieldQuery]] = {}
```

Without the default `mypy` will complain asking for some type to be specified.

> shouldn't it be a more specific type?

I experimented with this, for example I tried using `object` instead:

```
beets/dbcore/types.py:64: error: Incompatible types in assignment (expression has type "type[SubstringQuery]", variable has type "type[FieldQuery[object]]")  [assignment]
beets/dbcore/types.py:163: error: Incompatible types in assignment (expression has type "type[NumericQuery]", base class "Type" defined the type as "type[FieldQuery[object]]")  [assignment]
beets/dbcore/types.py:244: error: Incompatible types in assignment (expression has type "type[NumericQuery]", variable has type "type[FieldQuery[object]]")  [assignment]
beets/dbcore/types.py:274: error: Incompatible types in assignment (expression has type "type[SubstringQuery]", base class "Type" defined the type as "type[FieldQuery[object]]")  [assignment]
beets/dbcore/types.py:315: error: Incompatible types in assignment (expression has type "type[BooleanQuery]", base class "Type" defined the type as "type[FieldQuery[object]]")  [assignment]
```

This broke it, essentially. As far as I understood anything more specific than `Any` set as default ties to that type, so it kind of loses its generic properties.